admin/threads-seq.rs

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env -S cargo +nightly --quiet -Zscript
+
+//! # `admin/thread-seq N`
+//!
+//! This program prints a sequence of N integers for multithreaded
+//! performance testing.  The integers are numbers of threads to
+//! be sampled in a test.  The goal is to assist in graphing
+//! how per-thread throughput relates to concurrency.
+//!
+//! The sequence is (at most) length N, starts at 2, includes the
+//! number of CPU cores, and ends at 1.5 the number of CPU cores.
+//! It does not have repeated items.
+//!
+//! We exceed the number of cores specifically to see the
+//! "elbow" in the graph, when the number of threads tested
+//! exceeds the number of cores.  (This is good because otherwise
+//! -- assuming the software under test is perfectly scalable --
+//! the graph would be a straight line parallel with the x axis.)
+
+use std::{cmp, env, error, str::FromStr, thread};
+
+fn main() -> Result<(), Box<dyn error::Error + Send + Sync + 'static>> {
+    let mut args = env::args();
+    args.next(); // skip argv[0]
+    let count = args
+        .next()
+        .map(|c| usize::from_str(&c))
+        .unwrap_or(Ok(16))?;
+
+    let cpus = env::var("CPU_COUNT")
+        .map(|c| usize::from_str(&c))
+        .unwrap_or_else(|_| {
+            Ok(thread::available_parallelism()
+                .unwrap()
```

Comment:
Nit: also `?` this?

Reply:
done (and also used nonzerousize everywhere)